src/world/GameWorldGame.cpp

```diff
@@ -1315,6 +1315,31 @@ void GameWorldGame::ConvertMineResourceTypes(Resource::Type from, Resource::Type
     }
 }
 
+void GameWorldGame::FillWaterEverywhere() 
```

Comment:
I would move ALL the water code here. So name it e.g. `PlaceAndFixWater`, check if there is water on the node or we want water everywhere and then replace it with the appropriate amount of water (as in the MapLoader).
Then you can remove the current code from MapLoader where it changes the amount of water depending on terrain and simply place water there.
And while you are on it: Why not check ALL terrains around the point for unsuitable terrain or e.g. start with 7 water and reduce (to 0 if required) if any terrain around can't hold that much water. You can use `GetRightTerrain` and iterate over the 6 directions (see e.g. `IsSeaPoint`)

Reply:
I moved the code as discussed. 
I hope my refactoring for IsSeaPoint and IsWaterPoint is acceptable. i didnt want to copy & paste the evaluation so i refactored to a more flexible way to evaluate the surrounding terrain.